Fix issue with historgram statistic set being converted by duration

williedoran opened the pull request:
removed covert by duration and added test for historgram min,max and sum reporting in statistic set

azagniotov commented on src/main/java/io/github/azagniotov/metrics/reporter/cloudwatch/CloudWatchReporter.java:
```diff
@@ -242,7 +242,7 @@ private void processHistogram(final String metricName, final Histogram histogram
             }
 
-            stageMetricDatum(builder.withStatisticSet, metricName, snapshot, StandardUnit.None, "snapshot-summary", metricData);
+            stageHistorgramMtricStatisticSet(builder.withStatisticSet, metricName, snapshot, StandardUnit.None, "snapshot-summary", metricData);
```
Typo in: `stageHistorgramMtricStatisticSet`

azagniotov commented on src/test/java/io/github/azagniotov/metrics/reporter/cloudwatch/CloudWatchReporterTest.java:
```diff
@@ -309,8 +324,11 @@ private MetricDatum metricDatumByDimensionFromCapturedRequest(final String dimen
-                        .filter(metricDatum -> metricDatum.getDimensions()
-                                .contains(new Dimension().withName("Type").withValue(dimensionValue)))
+                        .filter(metricDatum -> {
+                            System.out.println(metricDatum.getDimensions());
```
Forgot the `System.out.println` due to debugging?

azagniotov commented:
Thank you! Left some comments

williedoran commented:
Addressed the comments and added another test for a sliding window histogram to sanity check that subsequent report methods correctly update the min/max etc

azagniotov commented on src/test/java/io/github/azagniotov/metrics/reporter/cloudwatch/CloudWatchReporterTest.java:
```diff
@@ -289,6 +291,56 @@ public void shouldReportExpectedHistogramStdDevAsIs() throws Exception {
+        slidingWindowHistogram.update(100);
+        slidingWindowHistogram.update(5);
+        slidingWindowHistogram.update(6);
+        reporterBuilder.withStatisticSet().build().report();
```
I am curious: did you intentionally re-build the `CloudWatchReporter` before reporting the 2nd time? Why not just `reporter.report();` without rebuilding - I guess it can be another test case?

azagniotov commented:
Thank you for your input with this!

On Tue, Aug 15, 2017 at 3:07 AM, williedoran <notifications@github.com>
wrote:

> Addressed the comments and added another test for a sliding window
> histogram to sanity check that subsequent report methods correctly update
> the min/max etc
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/azagniotov/codahale-aggregated-metrics-cloudwatch-reporter/pull/6#issuecomment-322430106>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA8azKjDPo0S3JAIfKrIOI2dH3GK-eU5ks5sYW3ggaJpZM4O2taK>
> .
>
